File: 2028/go/main.go
func missingRolls(rolls []int, mean int, n int) []int {
	m := len(rolls)
	total := mean * (m + n)
	miss := total
	for _, it := range rolls {
		miss -= it
	}
	if miss < n || miss > 6*n {
		return nil
	}
	return mum2arr(miss, n)
}
func mum2arr(total int, n int) []int {
	quotient, remainder := total/n, total%n
	arr := make([]int, n)
	for i := range arr {
		arr[i] = quotient
	}

	if remainder == 0 {
		return arr
	}

	for i := 0; remainder > 0 && i < n; i++ {
		temp := min(remainder, 6-arr[i])
		arr[i] += temp
		remainder -= temp
	}
	return arr
}
func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

File: 2028/python3/main.py
class Solution:
    def missingRolls(self, rolls: list[int], mean: int, n: int) -> list[int]:
        miss = mean * (n + len(rolls)) - sum(rolls)
        if miss > n*6 or miss < n:
            return []
        return self.num2arr(miss, n)

    def num2arr(self, total:int, n:int) -> list[int]:
        quotient, remainder = divmod(total, n)
        arr = [quotient for _ in range(n)]
        if remainder == 0:
            return arr
        i = 0
        while remainder >0:
            temp = 6 - arr[i]
            if temp >= remainder:
                arr[i] += remainder
                break

            arr[i] = 6
            remainder -= temp
            i += 1

        return arr
